llvm/lib/Support/Unix/Signals.inc

```diff
@@ -65,7 +65,7 @@
 #ifdef __APPLE__
 #include <mach-o/dyld.h>
 #endif
-#if HAVE_LINK_H
+#if __has_include(<link.h>)
```

Comment:
Does it need `#if defined(__has_include)` guard too?

Reply:
It shouldn't. `__has_include` is supported by the minimum compiler versions that LLVM is listed as supporting (https://godbolt.org/z/ejYM4o7ca).

Couldn't test Apple clang, but I'm guessing it will be fine too.

There are also several uses of `__has_include` in the project not wrapped in a `#if defined(__has_include)`.

Reply:
(nitpicking)
On a second thought, it might be guarded the same way the `dl_iterate_phdr` use is guarded, that is
```
#if ENABLE_BACKTRACES && defined(HAVE_BACKTRACE) &&                            \
    (defined(__linux__) || defined(__FreeBSD__) ||                             \
     defined(__FreeBSD_kernel__) || defined(__NetBSD__))
```
This would avoid inclusion of `<link.h>` when unnecessary.

Reply:
Thought about this. Since the include and its only use site are very far apart, I prefer to stick with the simple `__has_include` (C++17, should be in all supported compilers)